server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -345,6 +346,26 @@ public void testMaxScriptFieldsSearch() throws IOException {
         }
     }
 
+    public void testIgnoreScriptfieldIfSizeZero() throws IOException{
+        createIndex("index");
+        final SearchService service = getInstanceFromNode(SearchService.class);
+        final IndicesService indicesService = getInstanceFromNode(IndicesService.class);
+        final IndexService indexService = indicesService.indexServiceSafe(resolveIndex("index"));
+        final IndexShard indexShard = indexService.getShard(0);
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        
+        
+        searchSourceBuilder.scriptField("field" + 0,
+                new Script(ScriptType.INLINE, MockScriptEngine.NAME, CustomScriptPlugin.DUMMY_SCRIPT, Collections.emptyMap()));
+        
+        searchSourceBuilder.size(0);
+        try (SearchContext context = service.createContext(new ShardSearchLocalRequest(indexShard.shardId(), 1, SearchType.DEFAULT,
+                searchSourceBuilder, new String[0], false, new AliasFilter(null, Strings.EMPTY_ARRAY), 1.0f, true, null, null))) {
+                    assertEquals(0, context.scriptFields().fields().size());
```

Comment:
nit: please move indentation to the right so it alligns at four spaces in the try-block)

Reply:
sorry, do you mean shift `assertEquals` four spaces backwards?